src/Agent.Sdk/Util/LoggedSecretMasker.cs

```diff
@@ -0,0 +1,74 @@
+using Microsoft.TeamFoundation.DistributedTask.Logging;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Text.RegularExpressions;
+
+namespace Agent.Sdk.Util
+{
+    /// <summary>
+    /// Extended secret masker service, that allows to log origins of secrets
+    /// </summary>
+    public class LoggedSecretMasker : ISecretMasker
+    {
+        private ISecretMasker _secretMasker;
```

Comment:
Maybe inheritance would be better here? Seems like a lot of methods just call the ones from `ISecretMasker`.
If there are advantages to using composition here I'm all for it.

Reply:
Truly speaking I don't see any advantages of using inheritance here. In any way, it will require us to override the original methods of SecretMasker in this class, but also it will bring higher coupling of components since we will depend on the particular implementation of `ISecretMasker` interface.